XMonad/Util/Hacks.hs

```diff
@@ -0,0 +1,67 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Util.Hacks
+-- Copyright   :  (c) 2020 Leon Kowarschick
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Leon Kowarschick. <thereal.elkowar@gmail.com>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- This module is a collection of random fixes, workarounds and other functions
+-- that rely on somewhat hacky implementations which may have unwanted sideeffects.
+--
+-- Import this module as qualified like so:
+--
+-- > import qualified XMonad.Util.Hacks as Hacks
+--
+-- and then use the functions you want as described in their respective documentation.
+--
+-----------------------------------------------------------------------------
+
+module XMonad.Util.Hacks
+  ( -- * Windowed fullscreen
+    -- $windowedFullscreenFix
+    windowedFullscreenFixEventHook
+  ) where
+
+
+import XMonad
+import Data.Monoid (All(All))
+import Control.Monad (when)
+
+
+-- $windowedFullscreenFix
+-- Windowed fullscreen describes the behaviour in which XMonad,
+-- by default, does not automatically put windows that request being fullscreened
+-- into actual fullscreen, but keeps them constrained
+-- to their normal window dimensions, still rendering them in fullscreen.
+--
+-- With chromium based applications like Chrome, Discord and others this
+-- can cause issues, where the window does not correctly see the size of the window
+-- when displaying the fullscreen content, thus cutting off the window content.
```

Comment:
I feel like now that we have settled on that this will be a hacks module, we can include the gifs you made somewhere around here.  Haddock supports this (c.f. [this library](https://hackage.haskell.org/package/reanimate-1.1.2.1/docs/Reanimate.html)) and it would (imo) make this feel a little bit more like a literate file full of useful little bits, instead of a coherent module

Reply:
Done, now including links to the gifs from this github issue in the docs. that _should_ be permanent enough, i hope?

Reply:
I would say so; it /is/ a hacks module, after all :)